### PR TITLE
Add credentials collectors explanation pages

### LINKED ISSUE #4212: Add credentials collectors explanation
### Add a new page 🆕

Add **Explanation** documentation for each credentials collector plugin, as well as credentials collectors in general. Add the new pages under `/features/`.

- [ ] Credentials collectors (general)
- [ ] Chrome
- [ ] Mimikatz
- [ ] SSH

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes #4212

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [ ] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: docs/content/features/credentials_collectors/_index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: "Credentials Collectors"
+chapter: true
+pre: "<i class='fas fa-key'></i> "
+---
+
+# Credentials Collectors
+
+The Infection Monkey has multiple ways to steal credentials from compromised
+machines:
```
Comment (mssalvatore) [inline docs/content/features/credentials_collectors/_index.md:10]: We should provide more context about _why_ you would want to steal credentials and what Monkey can do with stolen credentials. How are stole credentials used?

**Review by mssalvatore — COMMENTED** [review]
I think all of these pages need more information. What is mimikatz? Why would we steal credentials from chrome? How likely are those credentials to be useful in propagation? What's the risk if and SSH key is stolen?

File: docs/content/features/credentials_collectors/_index.md (hunk 1 above)
Comment (shreyamalviya) [reply docs/content/features/credentials_collectors/_index.md:10]: WDYT?
image

Comment (mssalvatore) [reply docs/content/features/credentials_collectors/_index.md]: What do you think about this format instead? 
image

Comment (shreyamalviya) [reply docs/content/features/credentials_collectors/_index.md]: We can do this with an empty h2 ("## ").
image

File: docs/content/features/credentials_collectors/chrome.md
Comment (mssalvatore) [inline docs/content/features/credentials_collectors/chrome.md:1]: This looks fantastic!

File: docs/content/features/credentials_collectors/ssh.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+title: "SSH Credentials Collector"
+draft: false
+description: "Collects SSH keys from Linux users"
+tags: ["credentials collector", "ssh", "linux"]
+pre: "<i class='fa fa-terminal'></i> "
+---
+
+## Description
+
+SSH public/private key pair are credentials that allow users to remotely access
```
Comment (shreyamalviya) [inline docs/content/features/credentials_collectors/ssh.md:11]:
```suggestion
SSH public/private key pairs are credentials that allow users to remotely access
```

File: docs/content/features/credentials_collectors/ssh.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+(SSH)](https://en.wikipedia.org/wiki/Secure_Shell). Stealing them could enable
+an attacker to:
+
+- Gain access to sensitive systems
+- Steal data
+- Move laterally through the network
+- Escalate their privileges
+- Establish a persistent presence
+
+While SSH key pair can be encrypted to mitigate this risk, many users skip this
```
Comment (shreyamalviya) [inline docs/content/features/credentials_collectors/ssh.md:22]:
```suggestion
While SSH key pairs can be encrypted to mitigate this risk, many users skip this
```

File: docs/content/features/credentials_collectors/ssh.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+step and trade security for convenience. This leaves the SSH keys vulnerable to
+theft. Infection Monkey's SSH Credentials Collector seeks out and steals
+unprotected SSH key pairs. After compromising a Linux host, the SSH Credentials
+Collector will locate the `$HOME/.ssh` directory for each user and attempt to
+steal unencrypted SSH key pair from it.
```
Comment (shreyamalviya) [inline docs/content/features/credentials_collectors/ssh.md:27]:
```suggestion
steal unencrypted SSH key pairs from it.
```

File: docs/content/features/credentials_collectors/ssh.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+The SSH Credentials Collector will attempt to steal any key pair matching all
+of the following criteria:
+
+- The key pair is stored in `$HOME/.ssh` for any user on the system.
+- The key pair is readable by the user running the Infection Monkey Agent.
+- The key pair is not encrypted.
+- The key pair is stored in in one of the supported formats (RSA, DSA, EC, or
```
Comment (shreyamalviya) [inline docs/content/features/credentials_collectors/ssh.md:35]:
```suggestion
- The key pair is stored in one of the supported formats (RSA, DSA, EC, or
```

**Comment by mssalvatore** [comment]
I expanded the explanation of Credentials Collectors:

image

I have two seemingly conflicting goals for this:

1. Provide context. More specifically, answer the question, "why steal credentials?" To that end, we also need to answer, "how are stolen credentials used?"
2. Keep this page concise and focused.

There may be a little bit too much detail about exploiter here, but I think it's important for users to understand how the stolen credentials are used. I think the solution would be to provide this information under `Self-propagating Agent` (Issue #4215) and just say, "for information about how stolen credentials are used ... blah blah ... see Self-propagating Agent". Since that page isn't ready yet, I think we should leave this text here. I've left a note in #4215 to extract/repurpose this text as part of that issue.

File: docs/content/features/credentials_collectors/_index.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+After the reconnaissance phase, the Agent will begin the propagation phase and
+attempt to compromise other hosts on the network. Exploiters are Infection
+Monkey plugins that attempt to spread copies of the Agent throughout the
+network. Some exploiters can use the credentials stolen by credentials
+collectors to gain access to other systems on the network. First, the exploiter
+will query the Monkey Island to retrieve credentials that were configured by
+the user and any credentials that were stolen by credentials collectors. Next,
+the exploiters will use the stolen credentials to attempt to authenticate with
+a target system. If authentication is successful, the exploiter will execute
+the Agent on the target system, spreading the infection throughout the network.
```
Comment (shreyamalviya) [inline docs/content/features/credentials_collectors/_index.md:37]: I'm still unsure if we should go into this much detail. I suggest we have a separate page explaining the different phases of execution which we could link to wherever required (probably each plugin type's explanation?). This isn't something that's specific to credentials collectors so it doesn't feel right to me here.

Comment (mssalvatore) [reply docs/content/features/credentials_collectors/_index.md:37]: I agree that it should be on separate pages, which is why I left notes in #4213 and #4215 to extract it. However, this is information users need to understand. It needs to be somewhere, and if we get interrupted in between now and when we complete #4213 and #4215, at least it's here.

**Pull request merged**